monero/wallet_client.go

```diff
@@ -135,14 +172,53 @@ func (c *walletClient) GetBalance(idx uint64) (*wallet.GetBalanceResponse, error
 	})
 }
 
+// WaitForTransReceipt waits for the passed monero transaction ID to receive
+// numConfirmations and returns the transfer information. While this function will always
+// wait for the transaction to leave the mem-pool even if zero confirmations are
+// requested, it is the caller's responsibility to request enough confirmations that the
+// returned transfer information will not be invalidated by a block reorg.
+func (c *walletClient) WaitForTransReceipt(req *WaitForReceiptRequest) (*wallet.Transfer, error) {
+	height, err := c.GetHeight()
+	if err != nil {
+		return nil, err
+	}
+
+	var transfer *wallet.Transfer
+
+	for {
+		transferResp, err := c.rpc.GetTransferByTxid(&wallet.GetTransferByTxidRequest{
+			TxID:         req.TxID,
+			AccountIndex: 0,
```

Comment:
will the account index always be zero?

Reply:
I'll fix it. Currently, it will always be zero, but the `Transfer` method that is called before it is passing the index as a parameter, so I'll update this function to do the same.